apps/javaVendor/src/TextInputComponent.java

```diff
@@ -0,0 +1,34 @@
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+
+public class TextInputComponent {
```

Comment:
this component seems like a great opportunity for some design patterns practice, we're duplicating a lot of code

Reply:
Yeah, I think I'll take a look at the factory pattern and ping you when its done

Reply:
Added the factory pattern to clean things up